foreshadow/transformers/base.py

```diff
@@ -440,91 +483,56 @@ def set_params(self, **params):
         self.keep_columns = params.pop("keep_columns", self.keep_columns)
         self.y_var = params.pop("y_var", self.y_var)
 
+        # Calls to override auto set the transformer instance
         self.override = params.pop("override", self.override)
-        if self.override is not None:
-            self._transformer = self._resolve(self.override)(**self.kwargs)
 
-        if self._transformer is not None:
+        if self.transformer is not None:
             valid_params = {
                 k.partition("__")[2]: v
```

Comment:
why is this needed?

Reply:
I didn't write this piece of code, it's a holdover from the initial version I left unchanged, I can dig into it later.